src/briefcase/commands/convert.py

```diff
@@ -0,0 +1,761 @@
+
+        # Copy changelog file
+        changelog_file = self.base_path / "CHANGELOG"
+        if not changelog_file.is_file():
+            self.logger.warning(
+                f"Changelog file not found in {self.base_path}. You should either create a new changelog file in"
+                f" {self.base_path} or rename an already existing changelog file to CHANGELOG."
+            )
+
+        # Copy tests or test entry script
+        test_path = self.base_path / test_source_dir
+        if test_path.exists():
+            test_entry_script = project_dir / "tests" / f"{module_name}.py"
+            copy2(
+                test_entry_script,
+                test_path / f"{module_name}.py",
+            )
+        else:
+            copytree(project_dir / "tests", test_path)
+
+    def convert_app(
+        self,
+        tmp_path: Path,
+        template: str | None = None,
+        template_branch: str | None = None,
+        project_overrides: dict[str, str] | None = None,
+        **options,
+    ) -> None:
+        """Run the wizard in a temporary directory and copy the necessary files into the
+        project.
+
+        :param tmp_path: Temporary path that should contain the files generated by
+            cookiecutter.
+        :param template: The cookiecutter template to use.
+        :param template_branch: The git branch that the template should use.
+        """
+        version, template, branch = self.get_version_and_template_info(
+            template, template_branch
+        )
+        self.input.prompt()
+        self.input.prompt("Let's setup an existing project as a Briefcase app!")
+        context = self.build_context(template, branch, version, project_overrides)
+        self.prompt_divider()  # close the prompting section of output
+
+        self.warn_unused_overrides(project_overrides)
+
+        self.logger.info()
+        self.logger.info(
+            f"Generating required files to set up {context['formal_name']!r} with Briefcase",
+            prefix=context["app_name"],
+        )
+
+        # Create the project files
+        self.generate_template(
+            template=template,
+            branch=branch,
+            output_path=tmp_path,
+            extra_context=context,
+            allow_fallback=version.dev is not None,
+        )
+
+        app_path = context["app_name"]
+        self.logger.info(
+            f"Application {context['formal_name']!r} has been generated. To run your application, type:\n"
+            "\n"
+            f"cd {app_path}\n"
+            "briefcase dev"
+        )
+
+        project_dir = tmp_path / context["app_name"]
+        self.migrate_necessary_files(
+            project_dir, context["test_source_dir"], context["module_name"]
+        )
+
+    def validate_pyproject_file(self) -> None:
+        """Cannot setup new app if it already has briefcase settings in pyproject."""
+        if (self.base_path / "pyproject.toml").exists():
```

Comment:
Is there any value in letting this validation pass if there is *no* `pyproject.toml` file? Most of the benefit provided by this wizard is provided by parsing PEP621 configuration data; if you don't have that... why not just make the existence of *a* pyproject.toml file a requirement for running this command? 

If nothing else, it means the project has been at least *superficially* modernised - the *existence* of a `pyproject.toml` file has an impact on how pip operates, so we even if it isn't fully PEP621 compliant, it won't be relying on some archaic legacy setup.py behavior that will be implicitly altered by adding a Briefcase configuration.